stl/inc/stacktrace

```diff
@@ -0,0 +1,348 @@
+// stacktrace standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _STACKTRACE_
+#define _STACKTRACE_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if !_HAS_CXX23
+#pragma message("The contents of <stacktrace> are available only with C++23 or later.")
+#else // ^^^ !_HAS_CXX23 / _HAS_CXX23 vvv
+
+#include <cstdint>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+using _Stacktrace_string_fill_callback = size_t(__stdcall*)(char* _Data, size_t _Size, void* _Context);
+
+using _Stacktrace_string_fill = size_t(__stdcall*)(
+    size_t _Size, void* _String, void* _Context, _Stacktrace_string_fill_callback _Callback);
+
+_EXTERN_C
+unsigned short __stdcall __std_stacktrace_capture(unsigned long _Frames_to_skip, unsigned long _Frames_to_capture,
+    void** _Back_trace, unsigned long* _Back_trace_hash) noexcept;
+
+// Some of these functions may throw
+// (they would propagate bad_alloc potentially thrown from string::resize_and_overwrite)
+
+// clang-format off
+void __stdcall __std_stacktrace_address_to_string(
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+void __stdcall __std_stacktrace_description(
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+void __stdcall __std_stacktrace_source_file(
+    const void* _Address, void* _Str, _Stacktrace_string_fill _Fill) noexcept(false);
+
+_NODISCARD unsigned int __stdcall __std_stacktrace_source_line(const void* _Address) noexcept;
+
+void __stdcall __std_stacktrace_to_string(const void* _Addresses, size_t _Size, void* _Str,
+    _Stacktrace_string_fill) noexcept(false);
+
+_NODISCARD void* __stdcall __std_stacktrace_get_debug_interface() noexcept;
+// clang-format on
+_END_EXTERN_C
+
+_STD_BEGIN
+inline size_t __stdcall _Stacktrace_string_fill_impl(
+    const size_t _Size, void* const _String, void* const _Context, const _Stacktrace_string_fill_callback _Callback) {
+    if (_Callback) {
+        struct _Stacktrace_string_fill_op {
+            size_t operator()(char* _Data, size_t _Size) {
+                return _Callback(_Data, _Size, _Context);
+            }
+
+            _Stacktrace_string_fill_callback _Callback;
+            void* _Context;
+        };
+
+        static_cast<string*>(_String)->resize_and_overwrite(_Size, _Stacktrace_string_fill_op{_Callback, _Context});

```

Comment:
Yes, it is simpler with lambda, but I had code bloat concern. Do you think simplification worth using lambda?

Reply:
Though I now agree that code bloat does not apply: even in /Od, where lambda is not inlined into call site, only one instantiation finally gets used. There's still some bloat of object files, but it may be not a big deal